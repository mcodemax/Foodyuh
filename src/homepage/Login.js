import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import './Login.scss';
import * as Yup from 'yup';
import { Formik, Form, Field, ErrorMessage } from 'formik';


//might need to implement isLoading banner until requests are finished
//for logging in
function Login({ login }) {
    const navigate = useNavigate();
    const [loginErrors, setLoginErrors] = useState([]);

    const initialValues = {
        username: "",
        password: "",
    };

    const validationSchema = Yup.object({
        username: Yup.string().min(1).required("Required"),
        password: Yup.string().min(1).required("Required"),
    });

    const onSubmit = async (values, {resetForm}) => {
        console.debug('In Login.js, onSubmit')
        console.log("Form Data: ", values);//comment out later so password not revealed
        
        const res = await login(values.username, values.password);

        if(res.success){
            navigate("/user/profile"); 
        }else{
            setLoginErrors(res.errors)//trouble shoot later how to actuallyy display these properly
            resetForm({});
        }
        //if login fail show error message and navigate back2Login

        // https://github.com/jaredpalmer/formik/issues/446
    };
    
    return (
        <div className="Login">
            { loginErrors.length ? 
                loginErrors.map(error => {
                    return (
                        <div className="Login-failed error">
                            <p>{error}</p>
                        </div> 
                    )
                })
                : ``
            }
            <Formik initialValues={initialValues}
                validationSchema={validationSchema}
                onSubmit={onSubmit}
            >
                <Form className={'Login-form'}>
                    <div className={'Login-form-username form'}>
                        <label htmlFor="username">Username</label>
                        <Field type="text" id="username" name="username" />
                        <p className='form-warning'><ErrorMessage name="username"  /></p>
                    </div>
                    <div className={'Login-form-password form'}>
                        <label htmlFor="password">Password</label>
                        <Field type="password" id="password" name="password" />
                        <p className='form-warning'><ErrorMessage name="password"  /></p>
                    </div>
                    <button type="submit">Submit</button>
                </Form>
            </Formik>
        </div>
    );
}

export default Login;