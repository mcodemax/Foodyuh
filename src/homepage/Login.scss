.Login {
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #c7d6e6;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  -webkit-font-smoothing: antialiased;
  box-sizing: border-box;
  width: auto;
  padding-top: 20px;
  padding-right: 15px;
  padding-left: 15px;
  margin-right: auto;
  margin-left: auto;
  max-width: 720px;
  text-align: center !important;

  .error p {
    font-family: Geneva, sans-serif;
    color: red;
    box-sizing: border-box;
    font-size: 10px;
    height: 12px;
  }

  .Login-form {
    display: flex;
    flex-direction: column;
    box-sizing: border-box;
    align-items: center;

    .form {
      display: flex;
      flex-direction: column;

      label {
        font-family: Verdana, Geneva, Tahoma, sans-serif;
        text-align: left;
      }

      input {
        border-radius: 0.25rem;
        font-family: Geneva, sans-serif;
      }

      .field {
        font-family: Geneva, sans-serif;
        display: flex;
        flex-direction: column;
        height: 90px;
      }

      .form-warning {
        font-family: Geneva, sans-serif;
        color: red;
        box-sizing: border-box;
        font-size: 10px;
        height: 12px;
      }
    }

    button {
      text-decoration: none;
      display: inline-block;
      text-align: center;
      vertical-align: middle;
      width: 100px;
      user-select: none;
      padding: 0.3rem 0.75rem;
      font-size: 1rem;
      border-radius: 0.25rem;
      color: #fff;
      background-color: #007bff;
      border-color: #007bff;
      margin-top: 1rem !important;
      font-weight: 600 !important;
      cursor: pointer;
    }
  }
}
