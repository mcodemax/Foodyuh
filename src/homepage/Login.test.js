import React from "react";
import { render } from "@testing-library/react";
import Login from './Login';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
   useNavigate: () => mockedUsedNavigate,
}));

it("renders without crashing", function () {
  render(<Login />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<Login />);
  expect(asFragment()).toMatchSnapshot();
});
