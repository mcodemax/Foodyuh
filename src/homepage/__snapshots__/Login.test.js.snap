// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Login"
  >
    <form
      action="#"
      class="Login-form"
    >
      <div
        class="Login-form-username form"
      >
        <label
          for="username"
        >
          Username
        </label>
        <input
          id="username"
          name="username"
          type="text"
          value=""
        />
        <p
          class="form-warning"
        />
      </div>
      <div
        class="Login-form-password form"
      >
        <label
          for="password"
        >
          Password
        </label>
        <input
          id="password"
          name="password"
          type="password"
          value=""
        />
        <p
          class="form-warning"
        />
      </div>
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
